File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        maxreact =0
        for i in range(len(heights)):
            left =i 
            right =i

            while left>0 and heights[i] <=heights[left-1]:
                left-= 1

            while right <len(heights)-1 and heights[i] <=heights[right+1]:
                right+=1

            area = heights[i]*(right-left+1)
            print(left ,right,area)
            maxreact = max(area,maxreact)
        return maxreact

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    
    diameterOfBinaryTree(root) {
        let diameter =0;
       function calculateDepth(root){
        if(root == null){return 0}
        const leftMax = calculateDepth(root.left);
        const rightMax = calculateDepth(root.right);
        diameter =Math.max(diameter,leftMax+rightMax);
        return 1+ Math.max(leftMax, rightMax);
    }
        calculateDepth(root, diameter);
        return diameter;
        
    }
    
}
